internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,313 @@
+			// create registry pod in kubernetes cluster
+			if pod, err = rp.Kubeclient.CoreV1().Pods(rp.pod.Namespace).Create(ctx,
+				rp.pod, metav1.CreateOptions{}); err != nil {
+				return fmt.Errorf("error creating registry pod: %v", err)
+			}
+			// assign rp.pod to the newly created pod
+			rp.pod = pod
+		} else {
+			return fmt.Errorf("error getting existing registry pod: %v", err)
+		}
+	} else {
+		// if an existing pod matching rp.pod.Name is found, assign rp.pod to the existing pod
+		rp.pod = tmpPod
+	}
+
+	return nil
+}
+
+// VerifyPodRunning calls checkPodStatus to verify pod status and returns the pod logs upon error
+func (rp *RegistryPod) VerifyPodRunning(ctx context.Context) error {
+	// upon creation of new pod, poll and verify that pod status is running
+	podCheck := wait.ConditionFunc(func() (done bool, err error) {
+		p, err := rp.Kubeclient.CoreV1().Pods(rp.pod.Namespace).Get(ctx,
+			rp.pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting pod %s: %w", rp.pod.Name, err)
+		}
+		return p.Status.Phase == corev1.PodRunning, nil
+	})
+
+	if err := rp.checkPodStatus(ctx, podCheck); err != nil {
+		podLogs, logErr := rp.GetLogs(ctx)
+		if logErr != nil {
+			return fmt.Errorf("error verifying pod creation: %v: and fetching logs: %v", err, logErr)
+		}
+		if viper.GetBool(flags.VerboseOpt) &&
+			podLogs != "" {
+			fmt.Println(podLogs)
+		}
+		return fmt.Errorf("registry pod did not become ready: %w", err)
+	}
+	return nil
+}
+
+// checkPodStatus polls and verifies that the pod status is running
+func (rp *RegistryPod) checkPodStatus(ctx context.Context, podCheck wait.ConditionFunc) error {
+	// poll every 200 ms until podCheck is true or context is done
+	err := wait.PollImmediateUntil(time.Duration(200*time.Millisecond),
+		podCheck, ctx.Done())
+	if err != nil {
+		return fmt.Errorf("error waiting for registry pod %s to run: %v", rp.pod.Name, err)
+	}
+
+	return err
+}
+
+// validate will ensure that RegistryPod required fields are set
+// and throws error if not set
+func (rp *RegistryPod) validate() error {
+	if len(strings.TrimSpace(rp.BundleImage)) < 1 {
+		return errors.New("bundle image cannot be empty")
+	}
+	if len(strings.TrimSpace(rp.DBPath)) < 1 {
+		return errors.New("registry database path cannot be empty")
+	}
+
+	if len(strings.TrimSpace(rp.Namespace)) < 1 {
+		return errors.New("pod namespace cannot be empty")
+	}
+
+	if len(strings.TrimSpace(rp.BundleAddMode)) < 1 {
+		return errors.New("bundle add mode cannot be empty")
+	}
+
+	if rp.BundleAddMode != SemverBundleAddMode && rp.BundleAddMode != ReplacesBundleAddMode {
+		return fmt.Errorf("invalid bundle mode %q: must be one of [%q, %q]",
+			rp.BundleAddMode, ReplacesBundleAddMode, SemverBundleAddMode)
+	}
+
+	return nil
+}
+
+// getPodName will return a string constructed from the bundle Image name
+func getPodName(bundleImage string) (string, error) {
+	image, err := container.ImageFromString(bundleImage)
```

Comment:
To avoid a k8s-adjacent dependency and prevent conflicts in case different repo's are being used for the same image name (I'm fairly sure this function is just trimming the repo from the path and removing everything after `:`), I suggest either hashing `bundleImage` and taking the hash's [base32 encoding](https://golang.org/pkg/encoding/base32/#pkg-variables), or using [`k8sutil.TrimDNS1123Label(k8sutil.FormatOperatorNameDNS1123(bundleImage))`](https://github.com/operator-framework/operator-sdk/blob/master/internal/util/k8sutil/k8sutil.go#L125-L139).

Reply:
you are right that the function is just trimming the repo from the path and removing everything after `:`, however it needs to be in a valid docker image format. 

`k8sutil.TrimDNS1123Label(k8sutil.FormatOperatorNameDNS1123(bundleImage))` seems to work for valid bundle image formats. However, the concern with using it is that it will proceed with normal execution for invalid bundle image formats such as `invalid` or `non/existent/image`, but using `container.ImageFromString(bundleImage)` will prevent those ones from proceeding further as it will validate and throw an error.

Reply:
`getPodName` shouldn't care about whether `bundleImage` is a valid image tag. If the image tag is invalid then the pod will not start, and an error will be returned anyway.

Reply:
We also need to consider bundle images that use SHA references. Something like `quay.io/myorg/myimage@sha256:2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824`

If we used the DNS formatting and trimming approach, what would the pod name end up looking like? It would be nice (though not strictly necessary) if all of the resources we create for a particular `run bundle` invocation have some sort of human-readable reference in their names back to the bundle that they're for.

To avoid conflicts, we could append a random suffix using apimachinery's [`rand.String`](https://godoc.org/k8s.io/apimachinery/pkg/util/rand#String)

Reply:
Well we probably want a deterministic name for `Get`ting the pod for logs and such. Perhaps just ignore the repo and use operator name + version, like in the package manifests registry creator.

Reply:
can we add a todo item for this and revisit in a follow-up after we decide? not sure if this should be blocking the merge.

Reply:
Switched to k8sutil functions to construct the pod name, and added a todo item.